lib/joepegs.ts

```diff
@@ -0,0 +1,104 @@
+import type { Address } from 'viem'
+import { getProxy } from "@/lib/proxy";
+import { JsonRpcProvider } from "ethers";
+import { JOE_API_KEY } from '@/lib/config';
+import { JSON_RPC_ENDPOINT } from "@/lib/config"
+
+
+// Utility function for joepegs API queries.
+async function fetchJSON(url: string) {
+  // TODO: Support pagination.
+  try {
+    const response = await fetch(url, {
+      headers: {
+        "x-joepegs-api-key": JOE_API_KEY,
+      }
+    });
+    if (!response.ok) {
+      throw new Error(`${url} - Response status: ${response.status}`);
+    }
+    const data = await response.json();
+    return data;
+  } catch (error: any) {
+    console.error(error.message);
+  }
+}
+
+
+// Get all the tokens for a given owner in a collection.
+export async function fetchCollectionItems(address: Address, collection: Address) : Promise<Array<{'tokenId': string}>> {
+  const url = `https://api.joepegs.dev/v3/users/${address}/items?collectionAddresses=${collection}`;
+  return await fetchJSON(url);
+}
+
+
+// Get all the transfer events for a given item.
+export async function fetchItemActivities(collection: Address, tokenId: string) : Promise<Array<{'fromAddress': string, 'toAddress': string, 'timestamp': string}>> {
```

Comment:
I suggest defining a type and then referencing here.

Seems like this is supposed to hold the same data as `Activity` interface defined below, so maybe change `Activity` type to match these property names and reference here?